src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -130,24 +130,28 @@ bool convert_precision(ov::pass::PassBase& pass,
     };
 
     auto convert_node_output_precision = [&](const std::shared_ptr<ngraph::Node>& node) {
+        // Handle case with Constants as they can have consumers from other nGraph Function object
+        const auto it = const_to_internal_output.find(node.get());
+        if (it != const_to_internal_output.end()) {
+            return fuse_type_to_constant(node, to, it->second);
+        }
```

Comment:
Based on[ the failing tests,](https://dev.azure.com/openvinoci/dldt/_build/results?buildId=441448&view=logs&j=2f2823e7-545a-5ed6-9d73-63b6cd69c54b&t=ffe810cc-1cea-5574-ed02-77983a59b25d&l=6537) additional conditions for this part can be considered:
`if (node->outputs().size == 0) return false;`
and 
 `if (node.get_output_element_type(0) == from)
   ...`
```suggestion
 if (node.get_output_element_type(0) == from) {
        const auto it = const_to_internal_output.find(node.get());
        if (it != const_to_internal_output.end()) {
            return fuse_type_to_constant(node, to, it->second);
        }
      }
```
But the root cause of failing tests and correctness of those changes need to be verified.

Reply:
I've approached it differently and it now works as expected.